src/features/island/buildings/components/building/tent/Tent.tsx

```diff
@@ -1,35 +1,84 @@
-import React from "react";
+import React, { useContext, useState } from "react";
 
 import tent from "assets/buildings/tent.png";
 
 import { PIXEL_SCALE } from "features/game/lib/constants";
 import { BuildingProps } from "../Building";
 import { BuildingImageWrapper } from "../BuildingImageWrapper";
+import { Modal } from "react-bootstrap";
+import { TentModal } from "./TentModal";
+import { NPC } from "features/island/bumpkin/components/NPC";
+import { interpretTokenUri } from "lib/utils/tokenUriBuilder";
+import { useActor } from "@xstate/react";
+import { Context } from "features/game/GameProvider";
+import classNames from "classnames";
+
+export const Tent: React.FC<BuildingProps> = ({
+  buildingId,
+  isBuilt,
+  onRemove,
+}) => {
+  const { gameService } = useContext(Context);
+  const [
+    {
+      context: { state, bumpkins },
```

Comment:
Could update this to only re-render changes to fields we care about ie. `buildings` and `bumpkins`.